File: src/gestorAplicacion/obras/Estanteria.java
package gestorAplicacion.obras;

import java.io.Serializable;

import java.util.ArrayList;

public class Estanteria implements Serializable {
	// ATRIBUTOS DE CLASE
	private static final long serialVersionUID = 1L;
	public static ArrayList<Estanteria> lista = new ArrayList<>();
	static protected int numeroEstanterias;

	// ATRIBUTOS INSTANCIA
	short numero;
	short piso;
	String[] limites = new String[2];
	private ArrayList<Publicacion> publicaciones= new ArrayList();

	// CONSTRUCTORES
	public Estanteria() {
	}

	public Estanteria(short numero, short piso, String[] limites) {
		this.numero = numero;
		this.piso = piso;
		this.limites = limites;
		numeroEstanterias++;
		lista.add(this);
	}

	// METODOS
	public String motrarInfo() {
		return "Estanteria número " + this.numero + " ubicada en el piso " + this.piso + " de la biblioteca.";
	}
	
	public String getPublicaciones() {
		String c= "";
		for(int i=0;i<this.publicaciones.size();i++) {
			c= c+ this.publicaciones.get(i).nombre+ "/n";
		}
		return c;
	}

	public void setPublicaciones(Publicacion publicaciones) {
		this.publicaciones.add(publicaciones);
	}
	
	
	//GETTERS SETTERS
	public static ArrayList<Estanteria> getLista() {
		return lista;
	}

	public static void setLista(ArrayList<Estanteria> lista) {
		Estanteria.lista = lista;
	}

	public static int getNumeroEstanterias() {
		return numeroEstanterias;
	}

	public static void setNumeroEstanterias(int numeroEstanterias) {
		Estanteria.numeroEstanterias = numeroEstanterias;
	}

	public short getNumero() {
		return numero;
	}

	public void setNumero(short numero) {
		this.numero = numero;
	}

	public short getPiso() {
		return piso;
	}

	public void setPiso(short piso) {
		this.piso = piso;
	}

	public String[] getLimites() {
		return limites;
	}

	public void setLimites(String[] limites) {
		this.limites = limites;
	}

}

File: Práctica 2/src/gestorAplicacion/obras/Estanteria.py
class Estanteria: #serializacion ?
    # serialversion ?

    #atributos de clase
    _lista = []
    _numeroEstanterias = 0

    #Constructor
    def __init__(self, numero=0, piso=0, limites=None, publicaciones=None) :
        self.numero = numero
        self.piso = piso
        if limites is None:
            self.limites=[]
        else:
            self.limites = limites
        if publicaciones is None:
            self._publicaciones = [] 
        else:
            self._publicaciones = publicaciones
        Estanteria._numeroEstanterias+=1
        Estanteria._lista.append(self)
        
            

    #Metodos
    @classmethod
    def eliminarEstanteria(cls,E):
        Estanteria._lista.remove(E)
        del E
    @classmethod
    def mostrarRegistros(cls):
        c = "Estanterias creadas "+ "\n"
        for x in Estanteria._lista :
            c= c + f'Estanteria (CP)#{x.getNumero()}, Piso#{x.getPiso()} \n'
        return c

    def mostrarInfo(self):
        return f'INFORMACION DE LA ESTANTERIA: \n Numero: {self.getNumero()} \n Limites: {self.getLimites()} \n Piso: {self.getPiso()}'

    def mostrarContenido(self):
        from Libro import Libro
        from Revista import Revista
        from Folleto import Folleto
        c=""
        for i,x in enumerate(self._publicaciones):
            c= c + str(i) + f'. {x.getNombre()} Codigo {x.getCodigo()} \n '
        if not self._publicaciones:
            c= "Estanteria vacia"
        elif isinstance(self._publicaciones[0], Libro):
            c="La Estanteria contiene los siguientes libros \n " + c
        elif isinstance(self._publicaciones[0], Revista):
            c="La Estanteria contiene las siguientes Revistas \n " + c
        elif isinstance(self._publicaciones[0], Folleto):
            c="La Estanteria contiene los siguientes Folletos \n " + c 
        return c
    
    def agregarPublicacion(self,p):
        publicaciones = self.getPublicaciones() + p
        self.setPublicaciones(publicaciones)
        for x in p:
            x.setEstanteria(self)
        
    
    @classmethod
    def getLista(cls):
        return Estanteria._lista
    @classmethod
    def getNumeroEstanterias(cls):
        return Estanteria._numeroEstanterias

    def getNumero(self):
        return self.numero
    def setNumero(self,a):
        self.numero = a

    def getPiso(self):
        return self.piso
    def setPiso(self,a):
        self.piso = a

    def getLimites(self):
        return self.limites
    def setLimites(self,a):
        self.limites= a

    def getPublicaciones(self):
        return self._publicaciones
    def setPublicaciones(self,a):
        self._publicaciones = a